### PR TITLE
[SMALLFIX] Remove AlluxioExceptionType

### PR DESCRIPTION
**Pull request by yupeng9**
The enum `AlluxioExceptionType` is not necessary, because the exception class can use the class name to create the new instance via reflection.
By removing this enum, new exception classes are allowed to be defined outside `common` project.

### CODE REVIEW

File: core/common/src/test/java/alluxio/replay/ReplayCacheTest.java [hunk 1]
```diff
@@ -237,7 +236,7 @@ public ThrowingCallable(Exception t) {
 
     @Override
     public Long call() throws AlluxioException {
-      // If it's an AlluxioException, don't wrap it in RuntimeException
+      // If it's a AlluxioException, don't wrap it in RuntimeException
```
Comment (aaudiber) [inline core/common/src/test/java/alluxio/replay/ReplayCacheTest.java]: this should be an, right?

File: core/common/src/main/java/alluxio/exception/AlluxioException.java [hunk 2]
```diff
@@ -65,13 +59,16 @@ public AlluxioTException toAlluxioTException() {
    * @param e the {link AlluxioTException} to convert to a {@link AlluxioException}
    * @return a {@link AlluxioException} of the type specified in e, with the message specified in e
    */
+  @SuppressWarnings("unchecked")
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/exception/AlluxioException.java]: could we suppress just the one line?

File: core/common/src/main/java/alluxio/exception/AlluxioException.java [hunk 3]
```diff
@@ -65,13 +59,16 @@ public AlluxioTException toAlluxioTException() {
...
   public static AlluxioException from(AlluxioTException e) {
-    AlluxioExceptionType exceptionType = AlluxioExceptionType.valueOf(e.getType());
-    Class<? extends AlluxioException> throwClass = exceptionType.getExceptionClass();
+    Class<? extends AlluxioException> throwClass;
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/exception/AlluxioException.java]: the declaration can move to the same line as the assignment

File: core/common/src/main/java/alluxio/exception/AlluxioException.java [hunk 4]
```diff
@@ -65,13 +59,16 @@ public AlluxioTException toAlluxioTException() {
...
     try {
-      return throwClass.getConstructor(String.class).newInstance(e.getMessage());
+      throwClass = (Class<? extends AlluxioException>) Class.forName(e.getType());
+      AlluxioException throwInstance =
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/exception/AlluxioException.java]: why create the variable throwInstance instead of returning directly?

**Comment by aaudiber** [comment]
Awesome change, it's always great to see 100 lines get shaved off while gaining functionality :)

**Comment by apc999** [comment]
+1 LGTM --- assuming you fix @aaudiber 's comment

**Comment by yupeng9** [comment]
Thanks for the comments. Addressed them. PTAL.

File: core/common/src/test/java/alluxio/replay/ReplayCacheTest.java [hunk 5]
```diff
@@ -256,7 +255,7 @@ public ThrowingCallableThrowsIOException(Exception t) {
 
     @Override
     public Long call() throws AlluxioException, IOException {
-      // If it's an AlluxioException or IOException, don't wrap it in RuntimeException
+      // If it's a AlluxioException or IOException, don't wrap it in RuntimeException
```
Comment (calvinjia) [inline core/common/src/test/java/alluxio/replay/ReplayCacheTest.java]: should this be reverted?

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**